### PR TITLE
Add lint check to example app & refactor pom configs

### PR DESCRIPTION
**Pull request by skiddykong**

### CODE REVIEW

File: example-context/example-service/example-event/example-event-processor/pom.xml [hunk 1]
```diff
@@ -35,74 +39,4 @@
...
-                            <goal>generate</goal>
-                        </goals>
-                        <phase>generate-sources</phase>
-                        <configuration>
-                            <generatorName>uk.gov.justice.raml.jms.core.JmsEndpointGenerator</generatorName>
-                            <sourceDirectory>${basedir}/src/raml</sourceDirectory>
-                            <outputDirectory>${project.build.directory}/generated-sources</outputDirectory>
-                            <basePackageName>uk.gov.justice.api</basePackageName>
-                            <includes>
-                                <include>**/*cakeshop-event-processor.messaging.raml</include>
```
Comment (matt-rich) [inline example-context/example-service/example-event/example-event-processor/pom.xml]: Is there anything special about these paths/values that mean they still need to be implemented in the per pom level (just this bit, not the entire config)?

File: example-context/example-service/example-event/example-event-processor/pom.xml [hunk 2]
```diff
@@ -35,74 +39,4 @@
...
-                            <goal>generate</goal>
-                        </goals>
-                        <phase>generate-sources</phase>
-                        <configuration>
-                            <generatorName>uk.gov.justice.services.clients.messaging.generator.MessagingClientGenerator</generatorName>
-                            <sourceDirectory>${basedir}/src/raml</sourceDirectory>
-                            <outputDirectory>${project.build.directory}/generated-sources</outputDirectory>
-                            <basePackageName>uk.gov.justice.api</basePackageName>
-                            <includes>
-                                <include>**/*public-event.messaging.raml</include>
```
Comment (matt-rich) [inline example-context/example-service/example-event/example-event-processor/pom.xml]: Same as above, are we losing anything as we are no longer including this specific path/value ?

Comment (skiddykong) [inline example-context/example-service/example-event/example-event-processor/pom.xml]: I think it is the same as other configurations, the public-event raml will be included in the normal *.messaging.raml pick up

Comment (skiddykong) [inline example-context/example-service/example-event/example-event-processor/pom.xml]: as above

File: example-context/example-service/example-event/example-event-processor/pom.xml (hunk 1 above)
Comment (skiddykong) [inline example-context/example-service/example-event/example-event-processor/pom.xml]: I don’t see any issues as the public.events.messaging.raml is picked up by *.messaging.raml includes in the parent pom

**Pull request merged**